crates/sqlexec/src/context/local.rs

```diff
@@ -217,12 +217,12 @@ impl LocalSessionContext {
         Ok(())
     }
 
-    pub fn register_env_reader(&mut self, env_reader: Box<dyn EnvironmentReader>) {
-        self.env_reader = Some(env_reader);
+    pub fn register_env_reader(&mut self, reader: Option<Arc<Box<dyn EnvironmentReader>>>) {
+        self.env_reader = reader;
```

Comment:
I believe this can be changed back if we make the above change to have `connect` accept `self` (which would remove the need for an Arc).

Reply:
I think we need the arc in order to allow it to be cloneable, and passed in from the bindings' package.

Reply:
I think the arc makes a lot of sense here because ideally you never want a situation where the environment reader, once set, is duplicated...